monitor/service.go

```diff
@@ -32,6 +32,39 @@ const (
 	MonitorRetentionPolicyReplicaN = 1
 )
 
+// tags provides thread-safe tag handling
+type tags struct {
+	mu   sync.RWMutex
+	tags map[string]string
+}
+
+// NewTags creates a new tags struct to use
+func newTags() *tags {
+	return &tags{
+		tags: make(map[string]string),
+	}
+}
+
+// Add adds a new tag to the tag collection
+func (t *tags) Add(key string, value interface{}) {
+	t.mu.Lock()
+	defer t.mu.Unlock()
+
+	t.tags[key] = fmt.Sprintf("%v", value)
+}
+
+// Tags safely returns a copy of the current tag mapping
+func (t *tags) Tags() map[string]string {
+	t.mu.RLock()
+	defer t.mu.RUnlock()
+
+	r := make(map[string]string)
```

Comment:
Just a note of interest, 1.18 gave us a nice map copy function.

https://pkg.go.dev/golang.org/x/exp/maps#Copy